### ISSUE TITLE
New biome change method cause server freeze

### ISSUE THREAD

**Issue by andris155** [ShaneBeee/GriefPreventionFlags#65]
### Description

New biome change method cause server freeze

### Steps to Reproduce

Change biome 

### Expected Behavior

No cause server freeze. Please restore old method!

### Errors / Screenshots

```
[13:44:01] [Server thread/INFO]: SIKOLY issued server command: /abandonallclaims
[13:44:11] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-186 (MC: 1.14.4) ---
[13:44:11] [Paper Watchdog Thread/ERROR]: The server has not responded for 10 seconds! Creating thread dump
[13:44:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:11] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[13:44:11] [Paper Watchdog Thread/ERROR]: The server is waiting on these chunks: []
[13:44:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:11] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[13:44:11] [Paper Watchdog Thread/ERROR]: 	PID: 32 | Suspended: false | Native: false | State: RUNNABLE
[13:44:11] [Paper Watchdog Thread/ERROR]: 	Stack:
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.WorldServer.notify(WorldServer.java:1447)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_14_R1.CraftWorld.refreshChunk(CraftWorld.java:523)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GPFlags.flags.FlagDef_ChangeBiome.changeBiome(FlagDef_ChangeBiome.java:29)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GPFlags.flags.FlagDef_ChangeBiome.onClaimDelete(FlagDef_ChangeBiome.java:83)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1062.execute(Unknown Source)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.EventExecutor$$Lambda$2622/0x0000000800d4a440.execute(Unknown Source)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaim(DataStore.java:670)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaim(DataStore.java:616)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaimsForPlayer(DataStore.java:1253)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.GriefPrevention.onCommand(GriefPrevention.java:1333)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_14_R1.CraftServer.dispatchCommand(CraftServer.java:739)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnection.handleCommand(PlayerConnection.java:1800)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:1608)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:47)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:5)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:23)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnectionUtils$$Lambda$4134/0x0000000801a99440.run(Unknown Source)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:1013)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:1006)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:1135)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:919)
[13:44:11] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.4/java.lang.Thread.run(Unknown Source)
[13:44:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:11] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH ---
[13:44:11] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:16] [Paper Watchdog Thread/ERROR]: --- DO NOT REPORT THIS TO PAPER - THIS IS NOT A BUG OR A CRASH  - git-Paper-186 (MC: 1.14.4) ---
[13:44:16] [Paper Watchdog Thread/ERROR]: The server has not responded for 15 seconds! Creating thread dump
[13:44:16] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:16] [Paper Watchdog Thread/ERROR]: Server thread dump (Look for plugins here before reporting to Paper!):
[13:44:16] [Paper Watchdog Thread/ERROR]: The server is waiting on these chunks: []
[13:44:16] [Paper Watchdog Thread/ERROR]: ------------------------------
[13:44:16] [Paper Watchdog Thread/ERROR]: Current Thread: Server thread
[13:44:16] [Paper Watchdog Thread/ERROR]: 	PID: 32 | Suspended: false | Native: false | State: RUNNABLE
[13:44:16] [Paper Watchdog Thread/ERROR]: 	Stack:
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.WorldServer.notify(WorldServer.java:1448)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_14_R1.CraftWorld.refreshChunk(CraftWorld.java:523)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GPFlags.flags.FlagDef_ChangeBiome.changeBiome(FlagDef_ChangeBiome.java:29)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GPFlags.flags.FlagDef_ChangeBiome.onClaimDelete(FlagDef_ChangeBiome.java:83)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor1062.execute(Unknown Source)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.EventExecutor$$Lambda$2622/0x0000000800d4a440.execute(Unknown Source)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:80)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:545)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaim(DataStore.java:670)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaim(DataStore.java:616)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.DataStore.deleteClaimsForPlayer(DataStore.java:1253)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		me.ryanhamshire.GriefPrevention.GriefPrevention.onCommand(GriefPrevention.java:1333)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.command.PluginCommand.execute(PluginCommand.java:45)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:159)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//org.bukkit.craftbukkit.v1_14_R1.CraftServer.dispatchCommand(CraftServer.java:739)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnection.handleCommand(PlayerConnection.java:1800)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnection.a(PlayerConnection.java:1608)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:47)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PacketPlayInChat.a(PacketPlayInChat.java:5)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:23)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.PlayerConnectionUtils$$Lambda$4134/0x0000000801a99440.run(Unknown Source)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.TickTask.run(SourceFile:18)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeTask(IAsyncTaskHandler.java:136)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeNext(IAsyncTaskHandler.java:109)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.aX(MinecraftServer.java:1013)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.executeNext(MinecraftServer.java:1006)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.IAsyncTaskHandler.executeAll(IAsyncTaskHandler.java:95)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:1135)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		app//net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:919)
[13:44:16] [Paper Watchdog Thread/ERROR]: 		java.base@11.0.4/java.lang.Thread.run(Unknown Source)
```

### Server Information
* **Server version/platform:** 1.14.4/Paper-186
* **GriefPrevention Version:** 16.12.0-d4d7ac4
* **GriefPreventionFlags Version:** 5.6.3

**Comment by ShaneBeee** [comment]
Ill take a look into this, and see if I can fix the crash issue.

**Comment by ShaneBeee** [comment]
Can you please try this version and see if it solves the issue:
https://github.com/ShaneBeee/GriefPreventionFlags/releases/tag/5.6.4-Beta1

I tried to replicate and couldn't get my server to crash, but I did re-write the change biome method to refresh chunks substantially less than it was before.

**Comment by BiGUNMAN** [comment]
We have also witnessed this issue, from my impression that beta has helped, but there is still lag when unclaiming large claims.

**Comment by ShaneBeee** [comment]
Okay, maybe I'll create a setting in the config for auto-refreshing chunks. This way if users are experiencing lag they can disable that feature.

**Comment by ShaneBeee** [comment]
So I realized I made a stupid mistake.
When a claim is abandoned, the biome is reset, I should have made it check the flag instance. OOPS!

Try this version, it will only reset biomes for claims that have this flag. This should heavily reduce the lag caused by abandoning claims.
https://github.com/ShaneBeee/GriefPreventionFlags/releases/tag/5.6.4-Beta2

**Comment by ShaneBeee** [comment]
Fixed in [**5.6.4**](https://github.com/ShaneBeee/GriefPreventionFlags/releases/tag/5.6.4)
